examples/time_frequency/plot_time_frequency_global_field_power.py

```diff
@@ -0,0 +1,143 @@
+       vol. 108, 328-342, NeuroImage.
+.. [3] Efron B. and Hastie T. Computer Age Statistical Inference (2016).
+       Cambrdige University Press, Chapter 11.2.
+"""
+# Authors: Denis A. Engemann <denis.engemann@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import somato
+
+###############################################################################
+# Set parameters
+data_path = somato.data_path()
+raw_fname = data_path + '/MEG/somato/sef_raw_sss.fif'
+
+# let's explore some frequency bands
+iter_freqs = [
+    ('Theta', 4, 7),
+    ('Alpha', 8, 12),
+    ('Beta', 13, 25),
+    ('Gamma', 30, 45)
+]
+
+###############################################################################
+# We create average power time courses for each frequency band
+
+# set epoching parameters
+event_id, tmin, tmax = 1, -1., 3.
+baseline = None
+
+# get the header to extract events
+raw = mne.io.read_raw_fif(raw_fname, preload=False)
+events = mne.find_events(raw, stim_channel='STI 014')
+
+frequency_map = list()
+
+for band, fmin, fmax in iter_freqs:
+    # (re)load the data to save memory
+    raw = mne.io.read_raw_fif(raw_fname, preload=True)
+    raw.pick_types(meg='grad', eog=True)  # we just look at gradiometers
+
+    # bandpass filter and compute Hilbert
+    raw.filter(fmin, fmax, n_jobs=1)  # use more jobs to speed up.
+    raw.apply_hilbert(n_jobs=1, envelope=False)
+
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, baseline=baseline,
+                        reject=dict(grad=4000e-13, eog=350e-6), preload=True)
+    # remove evoked response and get analytic signal (envelope)
+    epochs.subtract_evoked()
+    epochs._data = np.abs(epochs.get_data())
+    # now average and move on
+    frequency_map.append(((band, fmin, fmax), epochs.average()))
+
+###############################################################################
+# Now we can compute the Global Field Power
+
+# We first estimate the rank as this data is rank-reduced as SSS was applied.
+# Therefore the degrees of freedom are less then the number of sensors.
+
+rank = raw.estimate_rank()
+print('The numerical rank is: %i' % rank)
+rng = np.random.RandomState(42)
+
+# Then we prepare a bootstrapping function to estimate confidence intervals
+
+
+def get_gfp_ci(average, rank=rank):
+    """get confidence intervals from non-parametric bootstrap"""
+    indices = np.arange(len(average.ch_names), dtype=int)
+    gfps_bs = list()
+    for bootstrap_iteration in range(2000):
+        bs_indices = rng.choice(indices, replace=True, size=len(indices))
+        gfp_bs = np.sum(average.data[bs_indices] ** 2, 0) / rank
+        gfps_bs.append(gfp_bs)
+    gfps_bs = np.array(gfps_bs)
+    gfps_bs = mne.baseline.rescale(gfps_bs, average.times, baseline=(None, 0))
+    ci_low = np.percentile(gfps_bs, 2.5, axis=0)
+    ci_up = np.percentile(gfps_bs, 97.5, axis=0)
```

Comment:
maybe. But the code I present here is at least more explicit. Let me at least avoid two calls to np.percentile.